### PR TITLE
Feature/sn 422 server side search for entity fields

### PR DESCRIPTION
**Pull request by AlexanderMakarov124**

### CODE REVIEW

**Comment by vasiliy-chefonov** [comment]
Update the readme as well

**Review by vasiliy-chefonov — CHANGES_REQUESTED** [review]

File: docs/SEARCH.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    3. Without
+    4. quotes
+
+## Search types
+
+Every property can have their own search type.
+
+Now the project have these search types:
+
+    * Starts With Case Sensitive;
```
Comment (vasiliy-chefonov) [inline docs/SEARCH.md:20]: Can we provide the examples of the SQL queries for each type?

File: src/Saritasa.NetForge.Infrastructure.EfCore/Services/EfCoreDataService.cs [hunk 2]
```diff
@@ -39,4 +46,212 @@ public IQueryable<object> GetQuery(Type clrType)
...
+    /// <summary>
+    /// Gets call of method similar to <see cref="string.Contains(string)"/> but case insensitive.
+    /// </summary>
+    /// <remarks>
+    /// Uses <see cref="Regex.IsMatch(string, string, RegexOptions)"/> with <see cref="RegexOptions.IgnoreCase"/>.
+    /// </remarks>
+    private static MethodCallExpression GetContainsCaseInsensitiveMethodCall(
+        MemberExpression propertyExpression, ConstantExpression searchConstant)
+    {
+        // entity => Regex.IsMatch(((entityType)entity).propertyName, searchWord, RegexOptions.IgnoreCase)
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Infrastructure.EfCore/Services/EfCoreDataService.cs:203]: Should we use `GetConvertedExpressionWhenPropertyIsNotString` here as well?

**Review by vasiliy-chefonov — APPROVED** [review]

File: docs/SEARCH.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# NetForge - Search
+
+## Search input on entity details page
+
+Every word in the input will become separate search entry. Phrases between quoutes: `"` or `'` will also become separate entries.
```
Comment (toanphamsaritasa) [inline docs/SEARCH.md:5]:
```suggestion
Every word in the input will become separate search entry. Phrases between quotes: `"` or `'` will also become separate entries.
```

**Review by toanphamsaritasa — APPROVED** [review]

**Pull request merged**